### PR TITLE
Document change-log tooling

### PR DESCRIPTION
**Pull request by prprabhu-ms**

### CODE REVIEW

**Comment by JamesBurnside** [comment]
Unassigned myself for load balancing (+plus I am heavily invested in this change so outside eyes on a doc review is a good thing)

File: docs/references/release-changelogs.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+At a high level, change-log maintenance happens in two steps:
+
+- Each change, in the form of a Pull Request to `main` in this repository adds a change-file - a file with a short description and type of the change.
+  - Developers generate these files by running `rush changelog` and CI enforces the existance of change files in Pull Requests.
+- The change-files are collected and summarized into the change-logs as part of the [release process](../releases/creating-a-release.md).
+
+## Under the hood
+
+Under the hood, our change-log tooling is built atop [beachball](https://microsoft.github.io/beachball/overview/getting-started.html). We do have a fair bit of customization / wrapping tooling to make it work with two release channels. The wrappers live in the [common/scripts/changelog/](../../common/scripts/changelog/) folder.
+
```
Comment (PorterNan) [inline docs/references/release-changelogs.md:18]: Would be nice to add a example + a folder structure here for better demonstration (something like before after, for both collecting and changelog

for example:
before changelog command
```
- BETA_DIR/
  - changelog1.json
- CHANGELOG_DIR/
  - changelog1.json
```

```
- BETA_DIR/
  - changelog1.json
  - changelog2.json (new)
- CHANGELOG_DIR/
  - changelog1.json
  - changelog2.json (new)
```

Comment (PorterNan) [reply docs/references/release-changelogs.md:18]: Also for before/after collecting

**Review by PorterNan — APPROVED** [review]

File: docs/references/release-changelogs.md (hunk 1 above)
Comment (PorterNan) [reply docs/references/release-changelogs.md:18]: It should be still working well without an example, but just feel that it took me some time to understand the logic when reading the doc + reading the code

**Review by anjulgarg — APPROVED** [review]

**Review by edwardlee-msft — APPROVED** [review]

**Pull request merged**